multicast/__main__.py

```diff
@@ -168,210 +185,390 @@ def NoOp(*args, **kwargs):
-	)
-	return parser
 
+	__prologue__ = """Python Multicast Receiver. Primitives for a listener for multicast data."""
+
+	@classmethod
+	def setupArgs(cls, parser):
+		"""Will attempt add send args.
+
+			Testing:
+
+			Testcase 0: First setup test fixtures by importing multicast.
+
+				>>> import multicast
+				>>> multicast.hear is not None
+				True
+				>>> multicast.hear.McastHEAR is not None
+				True
+				>>>
+
+			Testcase 1: main should return an int.
+				A: Test that the multicast component is initialized.
+				B: Test that the hear component is initialized.
+				C: Test that the main(hear) function is initialized.
+				D: Test that the main(hear) function returns an int 0-3.
+
+				>>> multicast.hear is not None
+				True
+				>>> multicast.__main__.main is not None
+				True
+				>>> tst_fxtr_args = ['''HEAR''', '''--deamon''', '''--port=1234''']
+				>>> (test_fixture, junk_ignore) = multicast.__main__.main(tst_fxtr_args)
+				>>> test_fixture is not None
+				True
+				>>> type(test_fixture) #doctest: -DONT_ACCEPT_BLANKLINE, +ELLIPSIS
+				<...int...>
+				>>> int(test_fixture) >= int(0)
+				True
+				>>> int(test_fixture) < int(4)
+				True
+				>>>
+
+			Testcase 2: setupArgs should not error given valid input.
+				A: Test that the multicast component is initialized.
+				B: Test that the __main__ component is initialized.
+				C: Test that the McastRecvHearDispatch class is initialized.
+				D: Test that the setupArgs function returns without error.
+
+				>>> multicast.__main__ is not None
+				True
+				>>> multicast.__main__.McastRecvHearDispatch is not None
+				True
+				>>> tst_fxtr_args = argparse.ArgumentParser(prog="testcase")
+				>>> multicast.__main__.McastRecvHearDispatch.setupArgs(parser=tst_fxtr_args)
+				>>>
+
+		"""
+		if parser is not None:
+			parser.add_argument("""--port""", type=int, default=_MCAST_DEFAULT_PORT)
+			__tmp_help = """local interface to use for listening to multicast data; """
+			__tmp_help += """if unspecified, any one interface may be chosen."""
+			parser.add_argument(
+				"""--iface""", default=None,
+				help=str(__tmp_help)
+			)
+			__tmp_help = """multicast groups (ip addrs) to bind to for the udp socket; """
+			__tmp_help += """should be one of the multicast groups joined globally """
+			__tmp_help += """(not necessarily joined in this python program) """
+			__tmp_help += """in the interface specified by --iface. """
+			__tmp_help += """If unspecified, bind to 224.0.0.1 """
+			__tmp_help += """(all addresses (all multicast addresses) of that interface)"""
+			parser.add_argument(
+				"""--group""", default=_MCAST_DEFAULT_GROUP,
+				help=str(__tmp_help)
+			)
+			parser.add_argument(
+				"""--groups""", default=[], nargs='*',
+				help="""multicast groups (ip addrs) to listen to join."""
+			)
+
+	def _help_deamon_dispatch(self, *args, **kwargs):
+		_useHear = False if "is_deamon" not in kwargs.keys() else kwargs["is_deamon"]
```

Comment:
Is there any supporting documentation for the claim that 
> Using `key not in dict` is more efficient than `key not in dict.keys()` because it avoids creating a new list of keys.
If no supporting documentation for the claim as a whole, instead I'd like to know where the official documentation on the python dictionary type's function `keys()` is.

Please also generate test code for the performance difference in python utilizing the `unittest` module for my review.

Reply:
- Is the python dict's keys() function consistent for all python 3.x ? please provide the earliest official documentation supporting the current behavior.

- Please include the previously mentioned supporting documentation with links and summarize the suggestion with this whole conversation in a tone intended for the PR summary.